modules/secrets-store-azure.adoc

```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+
+:_content-type: PROCEDURE
+[id="secrets-store-azure_{context}"]
+= Mounting secrets from an external secrets store to a CSI volume
+
+TODO: intro
+
+This procedure assumes that you are using Azure Key Vault as the secrets store provider.
+
+.Prerequisites
+
+* You have installed the {secrets-store-operator}. See _Installing the Secrets Store CSI Driver_ for instructions.
+* You have installed a secrets store provider. See the appropriate vendor documentation for instructions. This procedure assumes that you have installed the link:https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/[Azure Keyvault Provider].
+// TODO: Is there only a portion of these docs that they should follow? And any additional steps they need to take for OCP?
```

Comment:
@ropatil010 @knelasevero  Two questions about linking to the Azure docs...

1. This shows both installing the azure key vault provider and using the upstream secrets store CSI driver. Is there a particular section that they should follow just for the key vault provider? Would it be the "Deployment using Helm " section? What I see here looks different than what I see in the test case, so that leads me to my next question
2. Is there anything that users need to do in addition to whatever we link them to in the azure docs?

Reply:
I think you are going to need to:

https://redhat-internal.slack.com/archives/C05CXR1PVRD/p1688657080330429?thread_ts=1688654738.357779&cid=C05CXR1PVRD

In addition to the provider external docs, (before them, actually).

Reply:
The section would be using the deployment yamls. But I'm not thinking if it is worth pointing people into the external docs if we have custom steps. Maybe would be better to keep the users in and just provider our yamls with security contexts made for openshift. Instead of changing/adding policies to allow these pods to "be less secure".

Reply:
@ropatil010

Reply:
@knelasevero so you're saying that the custom step they need to do is this, correct?

```
oc adm policy add-scc-to-user privileged system:serviceaccount:openshift-cluster-csi-drivers:csi-secrets-store-provider-azure
```

I might reorder this then to be an actual step (instead of a prerequisite) and then add this command below. Let me know if you think we need to do something other than link to upstream.

Reply:
> I might reorder this then to be an actual step (instead of a prerequisite) and then add this command below. Let me know if you think we need to do something other than link to upstream.

We can start like that. 

> Also, what exactly is this step doing / why do they need to do it?

it is making sscsid service account privileged, so the user does not get errors while applying the DaemonSet from upstream not containing security contexts.